### PR TITLE
ARROW-7350: [Python] Decode parquet statistics as scalars

### PR DESCRIPTION
**Pull request by wjones127**
Thanks to Joris for pointing out we had this function in C++.

### CODE REVIEW

File: python/pyarrow/_parquet.pyx [hunk 1]
```diff
@@ -226,61 +229,12 @@ cdef _cast_statistic_raw_max(CStatistics* statistics):
...
-        elif not itype.is_signed() and itype.bit_width() == 64:
-            return int(np.int64(value).view(np.uint64))
-        else:
-            return value
-    else:
-        # No logical boxing defined
-        return value
+        shared_ptr[CScalar] min
+        shared_ptr[CScalar] max
+    StatisticsAsScalars(statistics[0], &min, &max)
```
Comment (lidavidm) [inline python/pyarrow/_parquet.pyx:236]: Check the status here?

Comment (lidavidm) [reply python/pyarrow/_parquet.pyx:236]: And nit, but maybe use `c_min` or something instead of shadowing a builtin

**Review by lidavidm — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
And thanks for the PR @wjones127 !

Unfortunately it seems this one caused some failures in one of the nightly integration builds: https://github.com/ursacomputing/crossbow/runs/6071471966?check_suite_focus=true 
The kartothek build is failing with errors that _seem_ related to this (but didn't actually verify this!)

From a quick look at the error message, this might be because we now return a datetime.date object instead of integer for date type?

**Comment by jorisvandenbossche** [comment]
We already have `min`/`max` and `min_raw`/`max_raw`, so I would like to avoid adding a third one to overcome this backwards compatibility issue ...

It's certainly also an option to see this as a "bug fix" (we shouldn't have used integers for the date type before) and ask kartothek to update their code. It's also for the lesser used date type (not for timestamp), and quite likely the behaviour here now also might have changes for other less used types that were not explicitly handled before (eg I don't how "interval" type would be handled?)

**Comment by lidavidm** [comment]
Hmm, fair. In that case we should test all the data types then just to make the behavior explicit.

**Comment by wjones127** [comment]
FWIW I considered it a bug that date statistics returned an integer. That's why I added tests for dates and decimals.

Would you like me to add tests for all other types in a new PR?

**Comment by jorisvandenbossche** [comment]
Ah, I missed that you added `date` already to the tests as well. 
Checking the parquet docs, I think the only other logical type that we don't yet explicitly handled is "interval" (https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#interval), but that is also not yet handled by the C++ `StatisticsAsScalars`, so this won't have changed in behaviour (and in general I don't know the support of mapping our own interval types to parquet interval). 

@wjones127 would you want to open an issue with kartothek to warn them about the upcoming change?

**Comment by jorisvandenbossche** [comment]
Thanks for opening that issue!

On the short term, we should also fix our nightly builds (either temporarily disabling them altogether, or ideally on skipping those failing tests). Opened https://issues.apache.org/jira/browse/ARROW-16262 for tracking that

**Pull request closed without merging**